File: ios/CloneInstagram/Core/Models/UserSummary.swift
import Foundation

/// Compte dans une liste d'abonnés ou un résultat de recherche, vu par moi (ADR-006).
struct UserSummary: Equatable, Identifiable {
    let id: String
    let username: String
    let fullName: String
    let isPrivate: Bool
    let avatar: ImageVariants?
    /// Je suis ce compte.
    var isFollowing: Bool
    /// Ce compte me suit.
    let followsMe: Bool
}

/// Une page d'une liste paginée par curseur (ADR-007) ; `nextCursor` est `nil` sur la dernière page.
struct UserPage: Equatable {
    let items: [UserSummary]
    let nextCursor: String?
}

/// Relation après un follow ou un unfollow, et compteur d'abonnés du compte visé.
struct FollowStatus: Equatable {
    let isFollowing: Bool
    let followerCount: Int
}

/// Liste d'un compte : ses abonnés ou ses abonnements.
enum FollowListKind: Hashable, CaseIterable {
    case followers
    case following
}

File: apps/api/src/modules/social/domain/user-summary.ts
import type { ImageVariantPaths } from '../../../shared/domain/image-variants.ts';

/** Compte dans une liste d'abonnés ou un résultat de recherche, vu par l'appelant. */
export interface UserSummary {
  id: string;
  username: string;
  fullName: string;
  isPrivate: boolean;
  avatarVariants: ImageVariantPaths | null;
  /** Relation avec l'appelant ; les deux valeurs sont `false` pour l'appelant lui-même. */
  relationship: { following: boolean; followedBy: boolean };
}

/** Résultat d'un follow ou d'un unfollow : relation, et compteur d'abonnés du compte visé. */
export interface FollowStatus {
  following: boolean;
  followerCount: number;
}
